### PR TITLE
Refactor entry string to tensor functions

### PR DESCRIPTION
**Pull request by AmishgaAlphonius**
### Description

The `entry_string_to_tensor` functions were duplicated; they were in `utilities.h` and `parameters.h`. In one case it was templated with `spacedim` and the other not. This PR merges both uses of the same function in `utilities.h`. Since these are small functions, they were inlined. Hence, they don't have any explicit instantiation.

While doing this, redundant `if` conditions were noticed in the `manifolds.cc`, these were replaced by a `for` loop.

### Checklist (will be removed when merged)
See [this page](https://chaos-polymtl.github.io/lethe/documentation/contributing.html#pull-requests) for more information about the pull request process.

Code related list:
- [x] All in-code documentation related to this PR is up to date (Doxygen format)
- [x] Lethe documentation is up to date
- [x] The branch is rebased onto master
- [x] Code is indented with indent-all and .prm files (examples and tests) with prm-indent
- [x] If parameters are modified, the tests and the documentation of examples are up to date
- [x] Changelog (CHANGELOG.md) is up to date if the refactor affects the user experience or the codebase

Pull request related list:
- [x] No other PR is open related to this refactoring
- [x] Labels are applied
- [x] There are at least 2 reviewers (or 1 if small feature) excluding the responsible for the merge
- [x] If this PR closes an issue or is related to a project, it is linked in the "Projects" or "Development" section
- [x] If any future works is planed, an issue is opened
- [x] The PR description is cleaned and ready for merge

### CODE REVIEW

File: include/core/utilities.h [hunk 1]
```diff
@@ -489,24 +489,96 @@ int
...
+template <int spacedim>
+inline Tensor<1, spacedim>
+value_string_to_tensor(const std::string &value_string)
+{
+  std::vector<std::string> vector_of_string(
+    Utilities::split_string_list(value_string));
+  std::vector<double> vector_of_double =
+    Utilities::string_to_double(vector_of_string);
+
+  AssertThrow(vector_of_double.size() == 3 || vector_of_double.size() == 2,
```
Comment (oguevremont) [inline include/core/utilities.h:516]:
```suggestion
  AssertThrow(vector_of_double.size() == spacedim,
```

File: include/core/utilities.h [hunk 2]
```diff
@@ -489,24 +489,96 @@ int
...
+              ExcMessage("Invalid string: " + value_string +
+                         ". This should be a two or three dimensional vector "
+                         "or point."));
```
Comment (oguevremont) [inline include/core/utilities.h:519]:
```suggestion
                         ". This should be a " + std::to_string(spacedim) + " dimensional vector "
                         "or point."));
```

File: include/core/utilities.h [hunk 3]
```diff
@@ -489,24 +489,96 @@ int
...
+  std::vector<std::string> vector_of_string(
+    Utilities::split_string_list(value_string_0));
+  Tensor<1, spacedim> output_tensor;
+
+  // The used parameter is a list of values
+  if (vector_of_string.size() > 1)
+    {
+      std::vector<double> vector_of_double =
+        Utilities::string_to_double(vector_of_string);
+      AssertThrow(vector_of_double.size() == 3 || vector_of_double.size() == 2,
```
Comment (oguevremont) [inline include/core/utilities.h:564]: Same sugestion as above

**Review by oguevremont — APPROVED** [review]
It's clear!

File: include/core/utilities.h (hunk 1 above)
Comment (AmishgaAlphonius) [reply include/core/utilities.h:516]: From what I've seen in the code (tests), sometime people leave the string at 2 components, when they want a 3 component rank 1 tensor also.
That's why I left it general. 
(same for other instances)

**Review by blaisb — APPROVED** [review]
Awesome, thanks for the nice refactor.

**Pull request merged**